src/Persistence/SQL.php

```diff
@@ -533,20 +533,27 @@ public function _typecastLoadField(Field $f, $value)
             } elseif (is_string($v)) {
                 // ! symbol in date format is essential here to remove time part of DateTime - don't remove, this is not a bug
                 $format = ['date' => '+!Y-m-d', 'datetime' => '+!Y-m-d H:i:s', 'time' => '+!H:i:s'];
-                $format = $f->persist_format ?: $format[$f->type];
+                if ($f->persist_format) {
+                    $format = $f->persist_format;
+                } else {
+                    $format = $format[$f->type];
+                    if (strpos($v, '.') !== false) { // time possibly with microseconds, otherwise invalid format
+                        $format = preg_replace('~(?<=H:i:s)(?![. ]*u)~', '.u', $format);
+                    }
+                }
 
                 // datetime only - set from persisting timezone
                 if ($f->type == 'datetime' && isset($f->persist_timezone)) {
                     $v = $dt_class::createFromFormat($format, $v, new $tz_class($f->persist_timezone));
-                    if ($v === false) {
-                        throw new Exception(['Incorrectly formatted datetime', 'format' => $format, 'value' => $value, 'field' => $f]);
+                    if ($v !== false) {
+                        $v->setTimeZone(new $tz_class(date_default_timezone_get()));
```

Comment:
should be `setTimezone` instead of `setTimeZone`
https://www.php.net/manual/en/datetime.settimezone.php

Reply:
Fixed in the another PR https://github.com/atk4/data/pull/505 - merge that PR first.